src/graphics.c

```diff
@@ -641,62 +680,72 @@ void graphics_draw_character( display_context_t disp, int x, int y, char ch )
 
     int depth = __bitdepth;
 
+    // setting default font if none was set previously
+    if( sprite_font.sprite == NULL )
+    {
+        graphics_set_default_font();
+    }
+
     /* Figure out if they want the background to be transparent */
     int trans = __is_transparent( depth, b_color );
 
+    const int sx = ( ch % sprite_font.sprite->hslices ) * sprite_font.font_width;
+    const int sy = ( ch / sprite_font.sprite->hslices ) * sprite_font.font_height;
+    const int ex = sx + sprite_font.font_width;
+    const int ey = sy + sprite_font.font_height;
+
+    const int tx = x - sx;
+    const int ty = y - sy;
+
     if( depth == 2 )
     {
         uint16_t *buffer = (uint16_t *)__get_buffer( disp );
+        uint16_t *sp_data = (uint16_t *)sprite_font.sprite->data;
 
-        for( int row = 0; row < 8; row++ )
+        for( int yp = sy; yp < ey; yp++ )
         {
-            unsigned char c = __font_data[(ch * 8) + row];
+            const register int run = yp * sprite_font.sprite->width;
 
-            for( int col = 0; col < 8; col++ )
+            for( int xp = sx; xp < ex; xp++ )
             {
+                const char c = sp_data[xp + run];
                 if( trans )
                 {
                     if( c & 0x80 )
```

Comment:
What does it mean `c & 0x80` in the context of a 16bpp or 32bpp font?

Reply:
Honestly I'm not even sure how that check works, but seemed to work correctly to grab the transparent pixel, so I left it there instead of trying to use a function to check.

Reply:
In the previous code, the font was 1bpp. So the code was reading one byte, shifting it left 8 times, and checking each time the highest bit (0x80) to check whether it was 0 or 1 (transparent or solid). 

In the new code, where the font is 16bpp or 32bpp, I would expect it to check the alpha component. In case of 16bpp it is a single bit, while for 32bpp fonts it would be a full byte. Does that make sense to you?